### ISSUE TITLE
Store/Update Bill API end_date and extension_date cant be null in request

### ISSUE THREAD

**Issue by jkano** [firefly-iii/firefly-iii#9147]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

Hi,

I noticed an issue when using the API for creating or updating a Bill (**POST /api/v1/bills** and **PUT /api/v1/bills/<id>**) vs creating same bill using Firefly frontend.

When I create or update a Bill for example sending the following data:

```json
{
  "name": "Test",
  "currency_id": "25",
  "currency_code": "COP",
  "amount_min": "1",
  "amount_max": "100000",
  "date": "2024-08-01T08:00:00-05:00",
  "end_date": null,
  "extension_date": null,
  "repeat_freq": "monthly",
  "skip": 0,
  "active": true,
  "notes": null,
  "object_group_id": null,
  "object_group_title": null
}
```

I get an error:

```json
{
    "message": "The end date is not a valid date. (and 3 more errors)",
    "errors": {
        "end_date": [
            "The end date is not a valid date.",
            "The end date must be a date after date."
        ],
        "extension_date": [
            "The extension date is not a valid date.",
            "The extension date must be a date after date."
        ]
    }
}
```

This means that end_date and extension_date can't be null according to the API response, but they should be able to be null as you can have a bill with no end date or extension date.

Also, in the frontend, I noticed that when creating/updating a Bill it says that the end date and extension date fields are **Optional fields**.

So, when I create/update a bill using the frontend and then request the bill info using GET /api/v1/bills:
```json
{
    "type": "bills",
    "id": "17",
    "attributes": {
        "created_at": "2024-08-18T12:56:54-05:00",
        "updated_at": "2024-08-18T12:56:58-05:00",
        "currency_id": "25",
        "currency_code": "COP",
        "currency_symbol": "COP$",
        "currency_decimal_places": 0,
        "name": "Test",
        "amount_min": "1",
        "amount_max": "100000",
        "date": "2024-08-18T00:00:00-05:00",
        "end_date": null,
        "extension_date": null,
        "repeat_freq": "monthly",
        "skip": 0,
        "active": true,
        "order": 1,
        "notes": null,
        "object_group_id": null,
        "object_group_order": null,
        "object_group_title": null,
        "next_expected_match": "2024-08-18T00:00:00-05:00",
        "next_expected_match_diff": "today",
        "pay_dates": [
            "2024-08-18T00:00:00-05:00"
        ],
        "paid_dates": []
    },
    "links": {
        "self": "https://firefly.jdkno.online/api/v1/bills/17",
        "0": {
            "rel": "self",
            "uri": "/bills/17"
        }
    }
}
```

As you can see the **end_date** and **extension_date** fields are **null** so these can be **null** but the API is rejecting requests with **null** on these fields.

### Debug information

Debug information generated at 2024-08-18 13:03:58 for Firefly III version **v6.1.18**.

<table><thead><tr><th colspan="2">System information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>Firefly III</td><td>6.1.18 / v2.1.0 / 24 (exp. 24)</td></tr><tr><td>PHP version</td><td>8.3.8 (64bits) / apache2handler / Linux x86_64</td></tr><tr><td>BCscale</td><td>12</td></tr><tr><td>Error reporting</td><td>Display: Off, reporting: ALL errors</td></tr><tr><td>Max upload</td><td>67108864 (64 MB)</td></tr><tr><td>Database drivers</td><td><strong>*mysql*</strong>, pgsql, sqlite, </td></tr><tr><td>Docker build</td><td><span>#</span>1104, base <span>#</span>85</td></tr></tbody></table><table><thead><tr><th colspan="2">Firefly III information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>Timezone</td><td>America/Bogota + America/Bogota</td></tr><tr><td>App environment</td><td>local, debug: true</td></tr><tr><td>Layout</td><td>v1</td></tr><tr><td>Logging</td><td>notice, stack / (empty)</td></tr><tr><td>Cache driver</td><td>file</td></tr><tr><td>Default language and locale</td><td>es_ES + equal</td></tr><tr><td>Trusted proxies</td><td>**</td></tr><tr><td>Login provider & user guard</td><td>eloquent / web</td></tr><tr><td>Login headers</td><td>N/A + N/A</td></tr><tr><td>Stateful domains</td><td></td></tr><tr><td>Last cron job</td><td>2024-08-17 05:00:00 (1 day ago)</td></tr><tr><td>Mailer</td><td>log</td></tr></tbody></table><table><thead><tr><th colspan="2">User-specific information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>User</td><td><span>#</span>1 of 1</td></tr><tr><td>User flags</td><td><span title="Has reconciled">:ledger:</span> <span title="Has subscriptions">:email:</span></td></tr><tr><td>Session start</td><td>2024-08-01 00:00:00</td></tr><tr><td>Session end</td><td>2024-08-31 23:59:59</td></tr><tr><td>View range</td><td>1M</td></tr><tr><td>User language</td><td>en_US</td></tr><tr><td>User locale</td><td>en_US</td></tr><tr><td>Locale(s) supported</td><td>en_US.utf8: :white_check_mark:<br>en_US.UTF-8: :white_check_mark:<br></td></tr><tr><td>User agent</td><td>Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/127.0.0.0 Safari/537.36</td></tr></tbody></table>

### Expected behaviour

On Bills API, end_date and extension_date can be null when making Create/Update requests.

### Steps to reproduce

1. Create a bill using front end (leave end date and extension date empty)
2. Get the bill information using API, notice the null on end_date and extension_date fields.
3. Create a bill using API (leave end date and extension date null)
4. API response is an error about wrong end_date and extension_date fields

5. Update the bill using front end (i.e. change name only)
6. Get the bill information using API, notice the null on end_date and extension_date fields.
7. Update the bill using API (i.e. name only, leave end date and extension date null)
8. API response is an error about wrong end_date and extension_date fields

**Comment by JC5** [comment]
Nice find, I'll check it out.

**Comment by JC5** [comment]
This will be fixed in 6.1.22 :)
